### PR TITLE
Add warnings and reasons

### LINKED ISSUE #11: Add warning / demerit system
Should allow staff to issue and look up warnings that have been given to people for use in decisions on bans etc

### LINKED ISSUE #12: Add reason to bans
Add a drop down to select from common reasons to the ban function

### PR DESCRIPTION
**Pull request by tellowkrinkle**
Adds a system for giving warnings with reasons to people accessible from where the `Ban` button used to be.  Bans have been moved to the warning page, and a reason is required for those as well.

Closes #11 
Closes #12

### CODE REVIEW

File: checkIn/templates/admin/warnings.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	<div class="row">
+	    <div class="panel panel-default">
+	        <div class="panel-heading">
+	            <a href="/admin/lookup?sid={{ warnee.sid }}" class="btn btn-primary"><i class="glyphicon glyphicon-chevron-left"></i></a>
+	        </div>
+	        <ul class="list-group">
+	            <li class="list-group-item">
+		            <h3>Warn {{ warnee.name }}</h3>
+		            <form action="/admin/warnings/{{ warnee.sid }}" method="post">
+						<label for="reasonInput">Reason</label>
```
Comment (tman0) [inline checkIn/templates/admin/warnings.html]: For the form here you should take a look at [Bootstrap's form documentation](https://getbootstrap.com/docs/3.4/css/#forms-horizontal) and wrap everything in `div.form-group`s to fix the spacing on the page.

File: checkIn/templates/admin/warnings.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		                <select name="category" id="category">
+							<option value=""{% if category != "Other" %} selected{% endif %} disabled hidden>Please select a category</option>
+			                <option value="Brought Prohibited Items">Brought Prohibited Items</option>
+			                <option value="Disorderly Conduct">Disorderly Conduct</option>
```
Comment (tman0) [inline checkIn/templates/admin/warnings.html]: * Safety glasses violation
* Improper tool use
* Food or drink in lab
* User agreement violation
* Blocking path of egress
* Damaging equipment/supplies
* Littering (or some nicer way to word leaving a mess)
* Improper attire

File: checkIn/checkIn.py [hunk 3]
```diff
@@ -219,6 +221,21 @@ class CardScan(Base):
 	def __repr__(self):
 		return "<CardScan %d at %s>" % (self.card, self.time)
 
+class Warning(Base):
```
Comment (tman0) [inline checkIn/checkIn.py]: Warning count should be visible on the user lookup page and maybe other places as well, maybe the user search page too

Comment (tellowkrinkle) [reply checkIn/checkIn.py]: I added it to the search page, where do you think I should put it on the lookup page?

Comment (tman0) [reply checkIn/checkIn.py]: Maybe just like a little badge on the side of the row that shows the number of warnings the person's received. Just so someone can see it at a glance whether or not someone routinely causes issues.

Comment (tellowkrinkle) [reply checkIn/checkIn.py]: I think that's what I was calling the search page?  Why don't you check the latest commit and see if it does what you wanted

**Pull request merged**